### ISSUE TITLE
Pagination query builder code inconsistency with relations pagination

### ISSUE THREAD

**Issue by ScSherifTarek** [laravel/framework#39807]
- Laravel Version: 8.40.*
- PHP Version: 7.4.8
- Database Driver & Version: MySQL 10.4.13-MariaDB

### Description:
I could notice inconsistency with how the pagination selection works when using relations compared to the normal eloquent access.

When using the pagination method of any of our models while specifying the columns to select using the `select` method it's going to only get these columns.

When using the pagination method of a relation (`BelongsToMany` in my example) while specifying the columns to select using the `select` method, it's going to get these columns in addition to `*` or simply all the columns of the related table every time.

### Steps To Reproduce:
1. Define a `BelongsToMany` relation between two models, for example a  many to many relation between users and roles.
2. Let's say the role would have this attributes: 
```
id, name, created_at, updated_at
```
3. Use this code to get the roles ids and names that contains the letter `a` paginated created:
```
Role::where('name', 'like', '%a%')->select(['id', 'names'])->paginate()
```
4. The following code is supposed get the roles **only** ids and names that are assigned for the user with id=1 paginated, but in action it's going to get all the roles attributes including created_at, and updated_at columns.
```
User::find(1)->roles()->select(['id', 'names'])->paginate()
```

Commenting out this [line](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php#L840) would make it works as expected.

**Comment by ScSherifTarek** [comment]
I could make it works on my side by specifying the columns I need to the `paginate` method 
```
User::find(1)->roles()->paginate(null, ['id', 'names'])
```

**Comment by kskrlinnorth2** [comment]
Same "bug" I just noticed in my application, but only when using it on relation defined by hasManyThrough, while hasMany works just fine with select()->paginate().

For example:
// items is relation on model2 using hasMany
$model2->items()->select('foo', 'bar')->paginate(10); // Select works, we get only foo and bar columns
// items is relation on model1 using hasManyThrough
$model1->items()->select('foo', 'bar')->paginate(10); // Ignores select and it gets all columns
// Select columns via pagiante method
$model1->items()->paginate(10, ['foo', 'bar']); // Now it works, but code layout is different, which could confuse new developers

Provided solution works, but why paginate method ignores select() in some cases, and in some cases it just works? It's an inconsistent behaviour imo.

Closing an issue and saying "use it like this, and not like this" is not solution. It's jumping over the problem and also, in the end, gets you code that doesn't look the same throughout the application.
